src/bootstrap/sanity.rs

```diff
@@ -155,6 +155,11 @@ than building it.
             continue;
         }
 
+        // Some environments don't want or need these tools, such as when testing Miri.
+        if env::var_os("BOOTSTRAP_SKIP_TARGET_SANITY").is_some() {
+            continue;
+        }
```

Comment:
There's other checks similar to this (e.g., the cmake check for msvc below). I think my recommendation would be to see if we can refactor the sanity checking as a whole so there's a x.py flag that disables it. I think skipping it entirely won't work today, but if we move some of the detection logic to config.rs that might fix it.

Reply:
That will probably require more digging into x.py than I will have time for, unfortunately.

Reply:
Alright. I can probably make an attempt to do that.

Reply:
I added another check of BOOTSTRAP_SKIP_TARGET_SANITY to the second loop that iterates all targets, which could help in the interim. (When writing this PR I did not notice there are two such loops.)